lib/docker.rb

```diff
@@ -119,6 +119,20 @@ def ping(connection = self.connection)
     connection.get('/_ping')
   end
 
+  # Determine if the server is podman or docker.
+  def podman?(connection = self.connection)
```

Comment:
I wonder if we can instead cache this value on the connection. Right now we're going to perform this check every time we perform certain requests, which makes another API call to the version endpoint. A connection certainly isn't guaranteed to be fixed to a single docker endpoint, but that would mean swapping the daemon listening on the socket out from underneath the client, which would be a pretty odd thing to do...

Reply:
@tlunter I'm certainly fine with that. Some people don't care for instance variables popping up everywhere but happy to do that if you'd like.

I agree that it would be super weird to swap it out in the middle of a program.

Reply:
Yes please. Otherwise everything else looks good to me.